folly/build/bootstrap-osx-homebrew.sh

```diff
@@ -10,41 +10,52 @@ cd "$BASE_DIR"
 
 # brew install alias
 brew_install() {
-    brew install $@ || brew upgrade $@
+    brew install "$@" || brew upgrade "$@"
 }
 
 # install deps
 install_deps() {
 	# folly deps
-	dependencies=(autoconf automake libtool pkg-config double-conversion glog gflags boost libevent xz snappy lz4 jemalloc openssl)
+	dependencies=(
+        boost
+        cmake
+        double-conversion
+        gflags
+        glog
+        jemalloc
+        libevent
+        lz4
+        openssl
+        pkg-config
+        snappy
+        xz
+    )
 
 	# fetch deps
-	for dependency in ${dependencies[@]}; do
-		brew_install ${dependency}
+	for dependency in "${dependencies[@]}"; do
+		brew_install "${dependency}"
 	done
 }
 
-# set env flags
-export_flags() {
-	# fetch opt dirs
-	OPT_GFLAGS=$(brew --prefix gflags)
-	OPT_OPENSSL=$(brew --prefix openssl)
-
-	# export flags
-	export LDFLAGS=-L${OPT_OPENSSL}/lib
-	export OPENSSL_INCLUDES=-I${OPT_OPENSSL}/include
-	export GFLAGS_LIBS=-L${OPT_GFLAGS}/lib
-	export GFLAGS_CFLAGS=-I${OPT_GFLAGS}/include
-}
-
-# now the fun part
 install_deps
-export_flags
-autoreconf -ivf
-./configure --disable-silent-rules --disable-dependency-tracking
+
+(
+    cd ..
+    mkdir -p _build
+    cd _build
+)
```

Comment:
If we are going to build in a subdir, then the steps below need to happen within that subdirectory. 
I think this closing bracket needs to move to the bottom of the file.

Reply:
Good catch. I pushed up the shellcheck fixes without fully testing them unfortunately as I was running off to a flight. I apologize for that.

You are certainly right that the lifetime of that bash subprocess would need to be for both the configure and make invocations.

I've refactored the script a bit to not require the `cd ..` and hence the 

```
(
cd ..
// ...
)
```

isn't needed. We now create a build directory in the top level dir, rather than using the previous `BASE_DIR` (folly/folly) and then doing `cd ..` to get us to `folly` top level.